### PR TITLE
Create Backbone login

### PR DESCRIPTION
**Pull request by DO2016**
Create Backbone login

### CODE REVIEW

File: showcase/admin.py [hunk 1]
```diff
@@ -11,19 +13,55 @@ class CurrencyAdmin(admin.ModelAdmin):
 
 @admin.register(Product)
 class ProductAdmin(admin.ModelAdmin):
-    list_display = ('name', 'description', 'is_published', 'date_published', 'price', 'currency')
+    list_display = ('name', 'description', 'is_published', 'date_published', 'price', 'currency', 'get_ings')
+
+    def get_ings(self, obj):
+        return "\n".join([i.name for i in obj.ings.all()])
```
Comment (vsdudakov) [inline showcase/admin.py]: return "\n".join(list(obj.ings.all().values_list('name'), flat=True))

File: showcase/admin.py [hunk 2]
```diff
@@ -11,19 +13,55 @@ class CurrencyAdmin(admin.ModelAdmin):
...
-    list_display = ('price', 'name', )
+    list_display = ('name', 'price', 'get_products',)
+    search_fields = ['name', 'price',]
+    list_filter = ('name', 'price',)
+    ordering = ['-name']
+    filter_vertical = ('products',)
+    form = IngredientAdminForm
+
+    def get_products(self, obj):
+        return "\n".join([p.name for p in obj.products.all()])
```
Comment (vsdudakov) [inline showcase/admin.py]: the same

File: showcase/api.py [hunk 3]
```diff
@@ -1,25 +1,207 @@
...
+        user_name = bundle.request.META['HTTP_USERNAME']
+        qs = qs.filter(username=user_name)
+        if key == qs[0].api_key.key:
+            mdl = qs.get(username=user_name)
+            mdl.is_active = True
+            mdl.save()
+            return qs
+
+    class Meta(BackboneCompatibleResource.Meta):
+        queryset = CustomUser.objects.all()
```
Comment (vsdudakov) [inline showcase/api.py]: CustomUser.objects.filter(is_active=True).select_related()

File: showcase/api.py [hunk 4]
```diff
@@ -1,25 +1,207 @@
...
+
+
+class ProductResource(BackboneCompatibleResource):
+    reviews = fields.ToManyField('showcase.api.ReviewResource', 'reviews', related_name='product', blank=True, null=True, full=True)
+    currency = fields.ForeignKey(CurrencyResource, 'currency', full=True)
+    allowed_methods_list = ['get']
+    #sum_ings_price = fields.DecimalField(readonly=True)
+
+    class Meta(BackboneCompatibleResource.Meta):
+        queryset = Product.objects.all_published()
```
Comment (vsdudakov) [inline showcase/api.py:98]: prefetch_related

Comment (vsdudakov) [reply showcase/api.py:98]: select_related

File: showcase/api.py [hunk 5]
```diff
@@ -1,25 +1,207 @@
...
+
+class OrderItemResource(BackboneCompatibleResource):
+    class Meta(BackboneCompatibleResource.Meta):
+        queryset = OrderItem.objects.all()
+        resource_name = 'OrderItemResource'
+        authentication = ApiKeyAuthentication()
+        authorization = Authorization()
+
+
+class OrderResource(ModelResource):
```
Comment (vsdudakov) [inline showcase/api.py:176]: remove

File: showcase/models.py [hunk 6]
```diff
@@ -1,19 +1,59 @@
...
 
 class ProductManager(models.Manager):
     def all_published(self):
         return self.select_related().filter(is_published=True).annotate(sum_ings_price = Sum('ings__price'))
 
 
+class OrderManager(models.Manager):
+    def _validate(self, date_text):
+        try:
+            if date_text != datetime.strptime(date_text, "%Y-%m-%d").strftime('%Y-%m-%d'):
```
Comment (vsdudakov) [inline showcase/models.py]: return date_text != datetime.strptime(date_text, "%Y-%m-%d").strftime('%Y-%m-%d')

**Review by vsdudakov — APPROVED** [review]

File: showcase/api.py (hunk 4 above)
Comment (DO2016) [reply showcase/api.py:98]: Done

File: showcase/models.py (hunk 6 above)
Comment (DO2016) [reply showcase/models.py]: Done

File: showcase/api.py (hunk 3 above)
Comment (DO2016) [reply showcase/api.py]: Done

File: showcase/admin.py (hunk 2 above)
Comment (DO2016) [reply showcase/admin.py]: Done

File: showcase/admin.py (hunk 1 above)
Comment (DO2016) [reply showcase/admin.py]: Done

**Pull request merged**